### PR TITLE
SSHDriver: Add port option

### PR DESCRIPTION
**Pull request by nefethael**
Signed-off-by: Vincent Prince <vincent.prince.fr@gmail.com>

### CODE REVIEW

File: labgrid/driver/sshdriver.py [hunk 1]
```diff
@@ -24,6 +24,7 @@ class SSHDriver(CommandMixin, Driver, CommandProtocol, FileTransferProtocol):
     """SSHDriver - Driver to execute commands via SSH"""
     bindings = {"networkservice": NetworkService, }
     keyfile = attr.ib(default="", validator=attr.validators.instance_of(str))
+    port = attr.ib(default=22, validator=attr.validators.instance_of(int))
```
Comment (Emantor) [inline labgrid/driver/sshdriver.py]: Remove the port attribute here, the port is available via the binding.

File: labgrid/driver/sshdriver.py [hunk 2]
```diff
@@ -114,11 +115,12 @@ def run(self, cmd, codec="utf-8", decodeerrors="strict"):
...
         (stdout, stderr, returncode)
         """
-        complete_cmd = "ssh -x {prefix} {user}@{host} {cmd}".format(
+        complete_cmd = "ssh -x {prefix} -p {port} {user}@{host} {cmd}".format(
             user=self.networkservice.username,
             host=self.networkservice.address,
             cmd=cmd,
-            prefix=self.ssh_prefix
+            prefix=self.ssh_prefix,
+            port=self.port
```
Comment (Emantor) [inline labgrid/driver/sshdriver.py]: needs a replacement as well.

File: labgrid/driver/sshdriver.py [hunk 3]
```diff
@@ -144,12 +146,13 @@ def get_status(self):
...
     def put(self, filename, remotepath=None):
-        transfer_cmd = "scp {prefix} {filename} {user}@{host}:{remotepath}".format(
+        transfer_cmd = "scp {prefix} {filename} -P {port} {user}@{host}:{remotepath}".format(
             filename=filename,
             user=self.networkservice.username,
             host=self.networkservice.address,
             remotepath=remotepath,
-            prefix=self.ssh_prefix
+            prefix=self.ssh_prefix,
+            port=self.port
```
Comment (Emantor) [inline labgrid/driver/sshdriver.py]: replace here too.

File: labgrid/driver/sshdriver.py [hunk 4]
```diff
@@ -167,11 +170,12 @@ def put(self, filename, remotepath=None):
     @Driver.check_active
     @step(args=['filename', 'destination'])
     def get(self, filename, destination="."):
-        transfer_cmd = "scp {prefix} {user}@{host}:{filename} {destination}".format(
+        transfer_cmd = "scp {prefix} -P {port} {user}@{host}:{filename} {destination}".format(
             filename=filename,
             user=self.networkservice.username,
             host=self.networkservice.address,
             prefix=self.ssh_prefix,
+            port=self.port,
```
Comment (Emantor) [inline labgrid/driver/sshdriver.py]: also `self.networkservice.port`

File: labgrid/driver/sshdriver.py [hunk 5]
```diff
@@ -189,8 +193,9 @@ def get(self, filename, destination="."):
 
     def _cleanup_own_master(self):
         """Exit the controlmaster and delete the tmpdir"""
-        complete_cmd = "ssh -x -o ControlPath={cpath} -O exit {user}@{host}".format(
+        complete_cmd = "ssh -x -o ControlPath={cpath} -O exit -p {port} {user}@{host}".format(
             cpath=self.control,
+            port=self.port,
```
Comment (Emantor) [inline labgrid/driver/sshdriver.py]: and here as well.

File: labgrid/driver/sshdriver.py [hunk 6]
```diff
@@ -54,8 +55,8 @@ def _start_own_master(self):
         sshpass = "sshpass -e " if self.networkservice.password else ""
         args = ("{}ssh -n {} -x -o ConnectTimeout=30 -o ControlPersist=300 "
                 "-o UserKnownHostsFile=/dev/null "
-                "-o StrictHostKeyChecking=no -MN -S {} {}@{}").format(
-                    sshpass, self.ssh_prefix, control,
+                "-o StrictHostKeyChecking=no -MN -S {} -p {} {}@{}").format(
+                    sshpass, self.ssh_prefix, control, self.port,
```
Comment (Emantor) [inline labgrid/driver/sshdriver.py]: Replace `self.port` with `self.networkservice.port`

**Comment by nefethael** [comment]
@jluebbe Hi, I updated PR to document NetworkService port new attribute, does it seem ok for you?

File: labgrid/driver/sshdriver.py [hunk 7]
```diff
@@ -144,12 +145,13 @@ def get_status(self):
     @Driver.check_active
     @step(args=['filename', 'remotepath'])
     def put(self, filename, remotepath=None):
-        transfer_cmd = "scp {prefix} {filename} {user}@{host}:{remotepath}".format(
+        transfer_cmd = "scp {prefix} {filename} -P {port} {user}@{host}:{remotepath}".format(
```
Comment (maximilianriemensberger) [inline labgrid/driver/sshdriver.py]: The port option must go before the filename.

**Comment by Emantor** [comment]
Merged manually, thanks.

**Pull request closed without merging**